src/ExpressionParser.php

```diff
@@ -586,16 +597,25 @@ public function parseFilterExpressionRaw($node, $tag = null)
         while (true) {
             $token = $this->parser->getStream()->expect(/* Token::NAME_TYPE */ 5);
 
-            $name = new ConstantExpression($token->getValue(), $token->getLine());
+            $name = $token->getValue();
             if (!$this->parser->getStream()->test(/* Token::PUNCTUATION_TYPE */ 9, '(')) {
                 $arguments = new Node();
             } else {
                 $arguments = $this->parseArguments(true, false, true);
             }
 
-            $class = $this->getFilterNodeClass($name->getAttribute('value'), $token->getLine());
+            $filter = $this->getFilter($name, $token->getLine());
+
+            $ready = true;
+            if (!isset($this->readyNodes[$class = $filter->getNodeClass()])) {
+                $this->readyNodes[$class] = (bool) (new \ReflectionClass($class))->getConstructor()->getAttributes(FirstClassTwigCallableReady::class);
+            }
+
+            if (!$ready = $this->readyNodes[$class]) {
+                trigger_deprecation('twig/twig', '3.12', 'Twig node "%s" is not marked as ready for passing a "TwigFilter" instead instead of its name; please make it ready and then flag its constructor with the #[FirstClassTwigCallableReady] attribute.', $class);
```

Comment:
```suggestion
                trigger_deprecation('twig/twig', '3.12', 'Twig node "%s" is not marked as ready for passing a "TwigFilter" instead of its name; please make it ready and then flag its constructor with the #[FirstClassTwigCallableReady] attribute.', $class);
```